File: Data Structures & Algorithms/k-closest-points-to-origin/submission-1.cpp
class Solution {
public:
    struct Compare {
        bool operator()(vector<int>& a, vector<int>& b){
            int aDist = pow(a[0], 2) + pow(a[1], 2);
            int bDist = pow(b[0], 2) + pow(b[1], 2);
            return aDist < bDist;
        }
    };

    vector<vector<int>> kClosest(vector<vector<int>>& points, int k) {
        priority_queue<vector<int>, vector<vector<int>>, Compare> pq;

        for(vector<int> v: points){
            pq.push(v);

            if(pq.size() > k) pq.pop();
        }

        vector<vector<int>> rslt;
        for(int i = 0; i < k; i++){
            rslt.push_back(pq.top());
            pq.pop();
        }

        return rslt;
    }
};

File: Data Structures & Algorithms/pow-x-n/submission-1.py
class Solution:
    def myPow(self, x: float, n: int) -> float:
        if n < 0:
            n = -n
            x = 1/x
        
        rslt = 1.0

        while n > 0:
            if n%2 == 1:
                rslt *= x
            
            x *= x
            n //= 2
        
        return rslt
